src/Rpc/NodeGuardService.cs

```diff
@@ -868,6 +870,42 @@ private bool ValidateTargetBalance(LiquidityRule rule)
         return true;
     }
 
+    public override async Task<GetAllUtxosResponse> GetAllUtxos(GetAllUtxosRequest request, ServerCallContext context)
+    {
+        List<FMUTXO> fmutxos;
+        // Check if the user sent a wallet id, else we get all utxos
+        if (request.WalletId == 0)
+        {
+            fmutxos = await _fmutxoRepository.GetAll();
+        }
+        else
```

Comment:
why are we filtering by wallet? 🤔

Reply:
the default case is that it brings all utxos no matter if the wallet is compromised. So i'm giving the option to the user to first fetch the available wallets and then get the utxos for those ones. I could also add a flag for bringing unarchived/uncompromised wallets

Reply:
Actually the functions `GetAvailableUtxos` and `GetAllUtxos` should be one function, but since `GetAvailableUtxos` is used by our backend to do settlements, I didn't want to modify it just yet. There is a difference between those two:

The `GetAvailableUtxos` function gets the utxos from the nbxplorer, then we remove the locked/frozen utxos. This means that that if there are utxos that haven't been tagged just yet, the caller to the function is going to get those utxos anyway.

The `GetAllUtxos` only returns utxos that have been already registered by us with the `RegusterUTXOsJob`.

- We have two things to do in the future: Add a flag to `GetAvailableUtxos` so we can choose to return only tagged utxos
- Merge both into the `GetAllUtxos` function and deprecate `GetAvailableUtxos`

Reply:
u're doing this in another task, right?

Reply:
yes

Reply:
https://github.com/Elenpay/NodeGuard/issues/382